.travis.yml

```diff
@@ -0,0 +1,12 @@
+
+services:
+  - docker
+
+script:
+  # Script for Building Docker Image and Run the Tests on TRAVIS CI
+  - docker build --target test --tag todo_app:test .
+  - docker run todo_app:test tests
+  - docker build --target test --tag todo_app:test_e2e .
```

Comment:
Note that there's no reason you need to rebuild this; the image will be identical to your todo_app:test image. If you do want to tag them separately for ease of reading, you could either tag them both in the initial build:
`docker build --target test --tag todo_app:test --tag todo_app:test_e2e`
or explicitly re-tag the image:
`docker tag todo_app:test todo_app:test_e2e`

Having said that, the build will be fully cached so while it will be slower than just tagging the existing image, it won't be a major issue

Reply:
Modified to tag in the same image build